### PR TITLE
Create markdown-it plugin to escape LaTeX

### LINKED ISSUE #435: Can we prevent collisions between Markdown and Mathjax?
See https://github.com/codidact/docs/issues/26 and the meta posts linked there.  Some Mathjax apparently has to be escaped because it's within Markdown, which confuses people periodically.  We can document it (the linked issue is in the docs repo), but if we can make it behave the way people are expecting, that'd be better.

### PR DESCRIPTION
**Pull request by MoshiKoi**
Should fix #435. However, I haven't fully tested this yet, so feedback is welcome.

### CODE REVIEW

**Comment by MoshiKoi** [comment]
It's been mostly tested and seems to work fine, so I'm opening it for review.

**Comment by MoshiKoi** [comment]
Okay, after fixing the most stupid mistake ever (return failure if validate is true), I have *actually* gotten all of my test posts rendering correctly.

File: app/assets/javascripts/latex_escape.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    // $$123$$ some text
+    //
+    // not math $$456$$
+    validate: match => match.indexOf('$$', 2) === match.length - 2.
+  },
+  '\\[': {
+    match: /\\\[[^]+?\\\]$/my   // \[...\]
+  },
+  'begin-end': {
+    match: /\\begin\{(.+)\}[^]+\\end\{\1\}$/my // \begin{...}...\end{...}
```
Comment (Taeir) [inline app/assets/javascripts/latex_escape.js:111]: Why is this one greedy in contrast to the others which are reluctant?

**Review by Taeir — COMMENTED** [review]
I learned a lot of new regex things reviewing this PR 😂 
Reluctant matching and how `[^]` works in Javascript regex.

I'm a bit curious though why the "sticky" flag is set on the regexes and what the effect of that is on your regex. Does that ensure it doesn't match inside of another match?

File: app/assets/javascripts/latex_escape.js (hunk 1 above)
Comment (MoshiKoi) [reply app/assets/javascripts/latex_escape.js:111]: No special reason, I probably just forgot. I'll change it

**Comment by MoshiKoi** [comment]
@Taeir 
> I'm a bit curious though why the "sticky" flag (`/y`) is set on the regexes and what the effect of that is on your regex. Does that ensure it doesn't match inside of another match?

> The `y` flag indicates that the regex attempts to match the target string only from the index indicated by the [`lastIndex`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/lastIndex) property (and unlike a global regex, does not attempt to match from any later indexes).

([MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/sticky))

**Pull request merged**